### PR TITLE
Upgrade dependencies.

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

**Comment by donraab** [comment]
@motlin Please can you rebase this and I can review/approve/merge? Thanks!

**Comment by motlin** [comment]
@donraab rebased

File: .idea/libraries/Maven__javax_annotation_javax_annotation_api_1_2.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (donraab) [inline .idea/libraries/Maven__javax_annotation_javax_annotation_api_1_2.xml:1]: Why is this being added?

File: .idea/libraries/Maven__net_java_dev_jna_jna_5_8_0.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (donraab) [inline .idea/libraries/Maven__net_java_dev_jna_jna_5_8_0.xml:1]: Why is this being added?

File: .idea/libraries/Maven__net_java_dev_jna_jna_platform_5_8_0.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (donraab) [inline .idea/libraries/Maven__net_java_dev_jna_jna_platform_5_8_0.xml:1]: Why is this being added?

File: .idea/libraries/Maven__net_sf_jopt_simple_jopt_simple_5_0_4.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (donraab) [inline .idea/libraries/Maven__net_sf_jopt_simple_jopt_simple_5_0_4.xml:1]: Why is this being added?

**Review by donraab — COMMENTED** [review]
Hi @motlin, most of the changes look fine where I could see upgrades to dependencies. The questions I have are on the new xml files being added. Can you explain the additions or remove them where they are not necessary? Thanks!

**Comment by motlin** [comment]
@donraab I wasn't sure, so I ran mvn dependency:tree to find where these dependencies are coming from. Here are the trimmed results.

```
org.eclipse.collections:eclipse-collections-code-generator-maven-plugin:maven-plugin:12.0.0-SNAPSHOT
+- org.apache.maven:maven-plugin-api:jar:3.9.0:provided
|  +- org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.5:provided
|  |  \- javax.annotation:javax.annotation-api:jar:1.2:provided

org.eclipse.collections:jcstress-tests:jar:12.0.0-SNAPSHOT
\- org.openjdk.jcstress:jcstress-core:jar:0.15:compile
   +- net.sf.jopt-simple:jopt-simple:jar:4.6:compile
   +- net.java.dev.jna:jna:jar:5.8.0:compile
   \- net.java.dev.jna:jna-platform:jar:5.8.0:compile
```

**Comment by donraab** [comment]
@motlin I'm not sure why the Inspections 2 build says pending. It looks like it passed on TC. Unless you have any objections, I will merge.

**Pull request merged**